lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -265,67 +265,207 @@ function _isAccountId(principal) {
     return (principal.length === 12 && /^\d+$/.test(principal));
 }
 
-function _checkPrincipal(requester, principal) {
-    if (principal === '*') {
-        return true;
-    }
-    // User in unauthenticated (anonymous request)
-    if (requester === undefined) {
+function _isRootUser(arn) {
+    if (!arn) {
         return false;
     }
-    if (principal === requester) {
+
+    // Vault returns the following arn when the account makes requests 'arn:aws:iam::295412255825:/master/',
+    // with an empty resource type ('user/' prefix missing).
+    const arns = arn.split(':');
+    const resource = arns.at(-1);
+
+    // If we start with '/' is because we have a empty resource type so we know it is a root account.
+    if (resource.startsWith('/')) {
```

Comment:
Missing protection against smaller arns:

```suggestion

    // Vault returns the following arn when the account makes requests 'arn:aws:iam::295412255825:/master/',
    // with an empty resource type ('user/' prefix missing).
    const arns = arn.split(':');
    const resource = arns.at(-1);

    if (arns.length < 6) {
        return false;
    }

    // If we start with '/' is because we have a empty resource type so we know it is a root account.
    if (resource.startsWith('/')) {
```